src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -115,16 +119,14 @@ static OpBase *_ReduceFilterToOpBase(ExecutionPlan *plan,
 }
```

Comment:
```
// In the case of a simple filter.
return NewFilterOp(plan, filter_root);
```

Why are we returning `NewFilterOp` I don't want to replace a filter operation which doesn't performs path filtering with a new version of itself.

Reply:
remember that this function is called only when a path pattern is present in the filter tree. The root cannot return filter op since it contains path pattern, so only a root of a nested subtree can return as a filter